packages/reactotron-server/src/server/index.ts

```diff
@@ -16,52 +17,43 @@ const argv = argsParser(process.argv.slice(2), {
 
 const config = getConfig(argv.config)
 
-pluginManager.loadPlugins()
-
-export const apolloServerInstance = async (app, httpServer) => {
-  const apolloServer = await createApolloServer()
-  apolloServer.applyMiddleware({ app })
-  apolloServer.installSubscriptionHandlers(httpServer)
-
-  return apolloServer
-}
+/**
+ * Configures and runs everything. There's a lot going on here, but the jist is this:
+ *
+ * - start a reactotron WebSocket server (the mobile apps talk to this thing)
+ * - start an http server using Express.JS
+ * - mount a graphql server
+ * - mount a react js app to host the reactotron ui
+ *
+ * @param config The configuration we'd like to use.
+ */
+export const bootUp = async (config: ReactotronServerConfig) => {
+  // load the reactotron plugins
+  pluginManager.loadPlugins()
+
+  // fireup reactotron-core-server
+  if (reactotron) {
+    reactotron.start(config.reactotronPort)
+  }
 
-export const httpServerInstance = () => {
+  // create an express app
   const app = express()
-  const httpServer = createServer(app)
-
-  app.use(express.static("public"))
-  app.get("/", (req, res) => {
-    const pluginScripts = pluginManager.getUiScripts();
 
-    res.send(
-      "<html>" +
-      "<head>" +
-      "<title>reactotron</title>" +
-      pluginScripts +
-      "<script src=\"bundle.js\"></script>" +
-      "</head>" +
-      "<body>" +
-      "<div id=\"root\" />" +
-      "</body>" +
-      "</html>"
-    )
-  })
+  // create an http server
+  const httpServer = createServer(app)
 
-  return { app, httpServer }
-}
+  // configure static routing for things like images & css that the reactotron web app might need
+  app.use(express.static(resolve(__dirname, "..", "..") + "/dist"))
 
-export const startServersListening = async (
-  { httpServer },
-  apolloServer = null,
-  reactotronServer = null,
-) => {
-  // If we have been given a reactotron server instance, start listening
-  if (reactotronServer) {
-    reactotronServer.start(config.reactotronPort)
-  }
+  // attach some endpoints to power our web app
+  const URL_PATH_TO_PLUGINS = "/reactotron-plugins.js"
+  app.get("/", Endpoints.createReactotronAppEndpoint(URL_PATH_TO_PLUGINS))
+  app.get(URL_PATH_TO_PLUGINS, Endpoints.createInjectPluginsEndpoint(pluginManager))
```

Comment:
I am not sure I follow how we are going to funnel multiple plugins scripts into a single endpoint like this? It might just be because I haven't tried it yet.

Reply:
This will be a 2nd endpoint that the web app will hit upon loading.  The `/reactotron-plugins.js` endpoint will then visit all the plugins available and collect up the javascript strings needed to power the plugins.

Now, what happens next, I haven't quite thought through.

I suspect that's what you're asking about.  So incase you're wondering, my answer is:  dunno yet.  :laughing:

Reply:
I think I came up with a few ideas over the weekend.